packages/common/src/components/Footer.jsx

```diff
@@ -7,23 +7,51 @@
  */
 
 import React from 'react'
+import { FooterLink } from '@kernel/common'
 
+const env = process.env.REACT_APP_DEPLOY_TARGET || 'PROD'
+const prefix = env === 'STAGING' ? 'staging.' : ''
 const defaults = {
   backgroundColor: 'bg-gray-900',
-  textColor: 'text-white'
+  textColor: 'text-white',
+  footerLinks: [
+    { href: 'https://' + prefix + 'explore.kernel.community', title: 'explorer' },
+    { href: 'https://' + prefix + 'unprofile.kernel.community', title: 'unprofile' },
+    { href: 'https://' + prefix + 'adventures.kernel.community', title: 'adventures' },
+    { href: 'https://' + prefix + 'wallet.kernel.community', title: 'portal' },
+  ]
```

Comment:
1. Let's break the URL-building logic out into a helper function that sits somewhere like `src/utils/index.js` or `src/utils/utils.js`, because the logic isn't footer-specific and because the footer shouldn't be too kernel-specific and because other things around the app may want this functionality too. The function should take a key like `'explorer'` and just return you the full URL of that app based on the current env.
2. You can export this function in `src/index.js` to expose it to the rest of the apps.
3. I'm not sure if the explicit `https://` is needed.. maybe try without?
4. Either way, you can use template strings to interpolate, instead of the `+`:
```
`${prefix}explore.kernel.community`
```
or:
```
`${env_prefix}${app_name}.kernel.community`
```

You can keep the `footerLinks` config here that just specifies the 4 links but calls on the util function to actually build the URLs.

Reply:
1. Done. Now using `getURL` from `src/utils/urls.js` which gets _app name_ as a param and returns the full URL for the corresponding env. `getURL` is used in `FooterLink` so that Footer stays agnostic.
2. Done.
3. Tried playing with it but it seems since in each `App.js` there's a `basename` defined, it is auto-added to the link as without the protocol the link is considered relative. I'll maybe play with it more later, but preferred to first push this as is if you do not object.
4. Done, thanks!

Reply:
Cool. I'd suggest keeping the `href` from before, so a single footer link config would be like:
```
{ href: getUrl('explorer'), title: 'explorer' }
```
So you'd import `getUrl` here, and then the `FooterLink` stays the way it was before, where it gets its `href` passed in and doesn't know about `getUrl` or anything else.

The reason is that this would allow the caller of `Footer` to pass in both kernel and non-kernel links.